worker/diskmanager/diskmanager.go

```diff
@@ -64,7 +64,8 @@ func doWork(listf ListBlockDevicesFunc, b BlockDeviceSetter, old *[]storage.Bloc
 		logger.Tracef("no changes to block devices detected")
 		return nil
 	}
-	logger.Infof("block devices changed: %#v", blockDevices)
+	logger.Infof("block devices changed")
+	logger.Tracef("new block devices: %#v", blockDevices)
```

Comment:
```suggestion
	logger.Infof("block devices changed")
	if logger.IsTraceEnabled() {
		logger.Tracef("new block devices: %#v", blockDevices)
	}
```

Reply:
> I don't quite understand why Simon suggested wrapping it in IsTraceEnabled. 

What's the point of serializing a set of devices if trace isn't enabled? You're still doing work to render a string. If `IsTraceEnabled` is slow in this path, then we should consider fix that.

Reply:
I don't think that it's useful to check against is TraceEnabled, since there is no compute required to get the log, and the trace function will do this check very early, before any argument processing.

https://github.com/juju/loggo/blob/ebdeb290b96116b911ea93cbe62107921610346e/logger.go#L293C1-L297C2

https://github.com/juju/loggo/blob/ebdeb290b96116b911ea93cbe62107921610346e/logger.go#L206

Reply:
Fair. Thanks for digging @gfouillet.